sdks/java/extensions/sql/zetasql/src/test/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlDialectSpecTest.java

```diff
@@ -4067,4 +4064,113 @@ public void testSimpleTableName() {
             Row.withSchema(singleField).addValues(15L).build());
     pipeline.run().waitUntilFinish(Duration.standardMinutes(PIPELINE_EXECUTION_WAITTIME_MINUTES));
   }
+
+  /** Tests for logical_and operation with different inputs. */
+  @RunWith(Parameterized.class)
+  public static class LogicalAndTests {
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> inputData() {
+      Object[][] data =
+          new Object[][] {{false, false, false}, {true, true, true}, {true, false, false}};
+      return Arrays.asList(data);
+    }
+
+    @Parameterized.Parameter(0)
+    private boolean firstValue;
+
+    @Parameterized.Parameter(1)
+    private boolean secondValue;
+
+    @Parameterized.Parameter(2)
+    private boolean logicalAndExpectedResult;
+
+    @Test
+    public void testLogicalAndAggregation() {
+      String sql =
+          "SELECT LOGICAL_AND(x) AS logical_and  FROM UNNEST(["
+              + firstValue
+              + ","
+              + secondValue
+              + "]) AS x";
+
+      ZetaSQLQueryPlanner zetaSQLQueryPlanner = new ZetaSQLQueryPlanner(config);
```

Comment:
@ibzib since `config` and `pipeline` are non-static variables, we cannot utilize them in the Parameterized Tests classes. However, if Beam would have been using JUNIT 5, we could have used [annotation based implementation](https://www.journaldev.com/21639/junit-parameterized-tests). I think, we need to keep the duplicate tests for now for different inputs.